### PR TITLE
feat: show message and domain hash for off-chain msgs

### LINKED ISSUE #4391: Add Message Hash of Ledger to UI
I would like to revive the not planned issue here: https://github.com/safe-global/safe-wallet-web/issues/1638. Can you please add the message hash displayed on the Ledger to the UI. This is super useful for verification really.

Please do this for message _and_ transactions as well please.

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves

Resolves #4391 

## How this PR fixes it
Adds two fields to off-chain msgs:
- message hash
- domain hash

Also makes those hashes always upper case to be easier to compare with what e.g. ledger displays

## How to test it
- Sign any off -chain msg and observe the new hashes

## Screenshots
Screenshot 2024-10-21 at 11 57 29

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/transactions/TxDetails/Summary/SafeTxHashDataRow/index.tsx [hunk 1]
```diff
@@ -16,12 +16,16 @@ export const SafeTxHashDataRow = ({
...
     chainId,
     verifyingContract: safeAddress,
   })
+    .slice(2)
+    .toUpperCase()}`
   const messageHash = safeTxData
-    ? TypedDataEncoder.hashStruct('SafeTx', { SafeTx: getEip712TxTypes(safeVersion).SafeTx }, safeTxData)
+    ? `0x${TypedDataEncoder.hashStruct('SafeTx', { SafeTx: getEip712TxTypes(safeVersion).SafeTx }, safeTxData)
+        .slice(2)
+        .toUpperCase()}`
```
Comment (iamacook) [inline src/components/transactions/TxDetails/Summary/SafeTxHashDataRow/index.tsx:28]: Nit: I'd add a comment _why_ this is in uppercase for future reference.

File: src/components/tx-flow/flows/SignMessage/SignMessage.tsx [hunk 2]
```diff
@@ -77,16 +77,27 @@ const createSkeletonMessage = (confirmationsRequired: number): SafeMessage => {
...
+const MessageHashField = ({
+  label,
+  hashValue,
+  capitalize,
+}: {
+  label: string
+  hashValue: string
+  capitalize?: boolean
+}) => {
+  const hash = capitalize ? `0x${hashValue.slice(2).toUpperCase()}` : hashValue
```
Comment (iamacook) [inline src/components/tx-flow/flows/SignMessage/SignMessage.tsx:89]: I'll leave it up to you, but we could create a helper to capitalise hex considering this is the third occurance.

**Review by iamacook — APPROVED** [review]

**Comment by liliya-soroka** [comment]
1. 2 times had different message hash ( research and steps are required)
2. 2 times signing message dialog from MM if the singing was canceled on Ledger or rejected in the MM:
Video : 

https://github.com/user-attachments/assets/229ba2c4-1529-4f4a-9da2-941782b9fd92

**Comment by katspaugh** [comment]
@liliya-soroka please come back to us on this one. This PR only adds new fields so unlikely to break anything else.

**Comment by liliya-soroka** [comment]
The both issues are still existing for me 
Yes, I am sure that I am checking the right fields , not signmessage and signmessagehash
image

**Comment by iamacook** [comment]
Marking this as a draft to revisit for older Safe versions.

**Comment by iamacook** [comment]
Superseeded by https://github.com/safe-global/safe-wallet-web/pull/4687.

**Pull request closed without merging**